x/crosschain/keeper/msg_server_vote_inbound_tx.go

```diff
@@ -57,74 +55,60 @@ import (
 // Only observer validators are authorized to broadcast this message.
 func (k msgServer) VoteOnObservedInboundTx(goCtx context.Context, msg *types.MsgVoteOnObservedInboundTx) (*types.MsgVoteOnObservedInboundTxResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-
-	observationType := observerTypes.ObservationType_InBoundTx
-	if !k.zetaObserverKeeper.IsInboundEnabled(ctx) {
-		return nil, types.ErrNotEnoughPermissions
-	}
-
-	// GetChainFromChainID makes sure we are getting only supported chains , if a chain support has been turned on using gov proposal, this function returns nil
-	observationChain := k.zetaObserverKeeper.GetSupportedChainFromChainID(ctx, msg.SenderChainId)
-	if observationChain == nil {
-		return nil, cosmoserrors.Wrap(types.ErrUnsupportedChain, fmt.Sprintf("ChainID %d, Observation %s", msg.SenderChainId, observationType.String()))
-	}
-	receiverChain := k.zetaObserverKeeper.GetSupportedChainFromChainID(ctx, msg.ReceiverChain)
-	if receiverChain == nil {
-		return nil, cosmoserrors.Wrap(types.ErrUnsupportedChain, fmt.Sprintf("ChainID %d, Observation %s", msg.ReceiverChain, observationType.String()))
-	}
-	tssPub := ""
-	tss, tssFound := k.zetaObserverKeeper.GetTSS(ctx)
-	if tssFound {
-		tssPub = tss.TssPubkey
-	}
-	// IsAuthorized does various checks against the list of observer mappers
-	if ok := k.zetaObserverKeeper.IsAuthorized(ctx, msg.Creator); !ok {
-		return nil, observerTypes.ErrNotAuthorizedPolicy
-	}
-
 	index := msg.Digest()
-	// Add votes and Set Ballot
-	// GetBallot checks against the supported chains list before querying for Ballot
-	ballot, isNew, err := k.zetaObserverKeeper.FindBallot(ctx, index, observationChain, observationType)
+
+	// vote on inbound ballot
+	// use a temporary context to not commit any ballot state change in case of error
+	tmpCtx, commit := ctx.CacheContext()
```

Comment:
Added it to make https://github.com/zeta-chain/node/blob/cfd27a705ab5d299e8eaa0e078bd6f2360a80ab2/x/crosschain/keeper/msg_server_vote_inbound_tx_test.go#L150 passing.
It is not necessary in practice now since the message would cancel the entire state transition but might become if we use the message internally. So it keep the specs: if the inbound finalized check doesn't pass, no ballot state change